blockchain/chain/src/store/chain_store.rs

```diff
@@ -51,16 +53,16 @@ pub enum HeadChange {
 /// This structure is threadsafe, and all caches are wrapped in a mutex to allow a consistent
 /// `ChainStore` to be shared across tasks.
 pub struct ChainStore<DB> {
-    publisher: RwLock<Publisher<HeadChange>>,
+    publisher: Arc<RwLock<Publisher<HeadChange>>>,
 
     // key-value datastore
     pub db: Arc<DB>,
 
     // Tipset at the head of the best-known chain.
-    heaviest: RwLock<Option<Arc<Tipset>>>,
+    heaviest: Arc<RwLock<Option<Arc<Tipset>>>>,
 
     // tip_index tracks tipsets by epoch/parentset for use by expected consensus.
-    tip_index: TipIndex,
+    tip_index: Arc<RwLock<TipIndex>>,
```

Comment:
I thought it was suggested the first time that the RPCState should just obtain mutable values that it needs to refrain from potentially blocking the blockstore. 

These both need to be in Sync Channel and RPCState, this is straying away from the block_store since we do not want to block that. It is a "taking what we need approach"